### ISSUE TITLE
Exception in file utils.py - function reverse_readline - python 3

### ISSUE THREAD

**Issue by helderfernandes1279** [certtools/intelmq#393]
When using python 3.4, when someone try to see the logs in intelmq-manager the following exception occurs in the function reverse_readline in file utils.py:

```
 File "/usr/local/lib/python3.4/dist-packages/intelmq-1.0.0-py3.4.egg/intelmq/lib/utils.py", line 243, in reverse_readline
    handle.seek(-offset, os.SEEK_END)
io.UnsupportedOperation: can't do nonzero end-relative seeks
```

Since python 3.2 that these arguments are not possible to use in the function seek, I've changed the code of the function and this fix works, here's the code:

```python
def reverse_readline(filename, buf_size=8192):
    with open(filename) as qfile:
        qfile.seek(0, os.SEEK_END)
        position = totalsize = qfile.tell()
        line = ''
        if buf_size < position:
            qfile.seek(totalsize-buf_size-1)
            char = qfile.read(1)
            while char != '\n':
                char = qfile.read(1)
            buffersize = totalsize-qfile.tell()
            number = totalsize-buf_size
        while position >= number:
            qfile.seek(position)
            next_char = qfile.read(1)
            if next_char == "\n":
                yield line[:: 1]
                line = ''
            else:
                line += next_char
            position -= 1
        yield line[:: 1]
```

Regards

**Comment by helderfernandes1279** [comment]
The code can be improved with the var names,  for example the var "number" can be replaced with "offset", this var store the value of the first possible char of a line in the range totalfilesize-buffersize

**Comment by helderfernandes1279** [comment]
There's a line missing in the code, the var number must be defined outside the "if".
Here's the code with the missing line:

def reverse_readline(filename, buf_size=8192):
----with open(filename) as qfile:
--------qfile.seek(0, os.SEEK_END)
--------position = totalsize = qfile.tell()
--------line = ''
--------number=0
--------if buf_size < position:
------------qfile.seek(totalsize-buf_size-1)
------------char=qfile.read(1)
------------while char!='\n':
------------------char=qfile.read(1)
------------buffersize=totalsize-qfile.tell()
------------number=totalsize-buf_size
--------while position >= number:

------------qfile.seek(position)
------------next_char = qfile.read(1)
------------if next_char == "\n":
----------------yield line[::-1]
----------------line = ''
------------else:
----------------line += next_char
------------position -= 1
--------yield line[::-1]

**Comment by sebix** [comment]
* The variable `buffersize` is assigned but never used.
* Your code adds an empty line at the end.
* Your code does not work if `buf_size` is smaller than the required amount of lines. It does not load the next buffer if needed.

Feel free to submit a pull request with a working code and ideally some unittests.

**Comment by helderfernandes1279** [comment]
The variable buffersize is assigned but never used:
    - You are right, I miss that
Your code does not work if buf_size is smaller than the required amount of lines. It does not load the next buffer if needed.
   - The main objective is to load the last part of the logs with the size of 8192, intelmq log files grows a lot, if you want to load the logs in the file, just remove all the code bellow (if buf_size < position)


Feel free to submit a pull request with a working code and ideally some unittests.
    At this moment I'm not able to do pull request, if someone need to work with intelmq-manager here's the correction:

def reverse_readline(filename, buf_size=8192):
    with open(filename) as qfile:
        qfile.seek(0, os.SEEK_END)
        position = totalsize = qfile.tell()
        line = ''
        offset = 0

        while position >= offset:
            qfile.seek(position)
            next_char = qfile.read(1)
            if next_char == "\n":
		if line[::-1]!='':
                   yield line[::-1]
                line = ''
            else:
                line += next_char
            position -= 1
        yield line[::-1]

**Comment by danje57** [comment]
Hi,

I've change the function in:

/usr/local/lib/python3.4/dist-packages/intelmq-1.0.0-py3.4.egg/intelmq/lib/utils.py

by this code

def reverse_readline(filename):
    with open(filename) as qfile:
        qfile.seek(0, os.SEEK_END)
        position = totalsize = qfile.tell()
        line = ''
        offset = 0
        while position >= offset:
           qfile.seek(position)
           next_char = qfile.read(1)
           if next_char == "\n":
               if line[::-1]!='':
                   yield line[::-1]
               line = ''
           else:
               line += next_char
        position -= 1
        yield line[::-1]


Now, I don't have the JSON error.

However, nothing appears in the monitor page :-/

I can't see any error in /var/log/httpd/error.log

**Comment by danje57** [comment]
Tried to debug by adding
def reverse_readline(filename):
   debugfile = open('/tmp/debug.txt', 'w')
   debugfile.write('TEST')
   debugfile.close()

But the debug.txt file is not created...

**Comment by danje57** [comment]
any news? please

**Comment by mauroasilva** [comment]
The bug was already fixed in the #435 pull request.
